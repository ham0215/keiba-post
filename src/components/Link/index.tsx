import { ReactNode } from 'react';
import NextLink from 'next/link';
import Link from '@mui/material/Link';

type Props = { href: string; prefetch?: boolean; children: ReactNode };

export default function LinkComponent({ href, prefetch = false, children }: Props) {
  return (
    <NextLink href={href} prefetch={prefetch} passHref>
      <Link color="inherit" underline="none">
        {children}
      </Link>
    </NextLink>
  );
}
