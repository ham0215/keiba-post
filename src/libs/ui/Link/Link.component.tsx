import { ReactNode } from 'react';
import NextLink from 'next/link';
import MuiLink from '@mui/material/Link';

type Props = { href: string; prefetch?: boolean; children: ReactNode };

export function Link({ href, prefetch = false, children }: Props) {
  return (
    <NextLink href={href} prefetch={prefetch} passHref legacyBehavior>
      <MuiLink color="inherit" underline="none">
        {children}
      </MuiLink>
    </NextLink>
  );
}
