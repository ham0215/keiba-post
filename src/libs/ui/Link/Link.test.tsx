import { Link } from './Link.component';
import { render } from '@testing-library/react';

describe('Link', () => {
  it('snapshot testing', () => {
    const { asFragment } = render(<Link href="/keiba">keiba</Link>);
    expect(asFragment()).toMatchSnapshot();
  });
});
