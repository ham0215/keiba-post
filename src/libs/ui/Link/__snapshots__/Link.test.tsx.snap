// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link snapshot testing 1`] = `
<DocumentFragment>
  <a
    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-6b04s2-MuiTypography-root-MuiLink-root"
    href="/keiba"
  >
    keiba
  </a>
</DocumentFragment>
`;
